test/js/test.hopscotch.js

```diff
@@ -155,6 +155,29 @@ describe('Hopscotch', function() {
       expect(hopscotch.getCurrStepNum()).toBe(1);
     });
 
+    it('should reject tour IDs that include invalid characters', function(){
+      try {
+        hopscotch.startTour({
+          id: '(this is a bad tour id!)',
+          steps: [
+            {
+              target: 'shopping-list',
+              orientation: 'left',
+              title: 'Shopping List',
+              content: 'It\'s a shopping list'
+            }
+          ]
+        });
+      }
+      catch (e) {
+        expect(true).toBeTruthy();
+        hopscotch.endTour();
+        return;
+      }
+      expect(false).toBeTruthy();
```

Comment:
Would be nice to add a comment here explaining that this test failed if it reached this point (should have thrown an exception). It's not very obvious from the first glance why this expect is here.